File: valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for ch in s:
            if (ch == '('):
                stack.append(')')
            elif (ch == '{'):
                stack.append('}')
            elif (ch == '['):
                stack.append(']')
            elif (len(stack)==0 or ch != stack.pop()):
                return False
            
        return len(stack)==0
        

File: valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        
        for (Character ch : s.toCharArray()){
            if(ch == '(') stack.push(')');
            else if(ch == '{') stack.push('}');
            else if(ch == '[') stack.push(']');
            else if(!stack.isEmpty() && ch == stack.peek()) stack.pop();
                else return false;         
        }
        
        return stack.isEmpty();
    }
}